README.md

```diff
@@ -2,6 +2,43 @@
 
 Silverstripe Search is designed to integrate easily with the Silverstripe CMS. This Software Development Kit (SDK) is a one-stop-shop for getting your application integrated.
 
+<!-- TOC -->
+* [🧰 Silverstripe Search SDK](#-silverstripe-search-sdk)
+  * [Tagged dependencies](#tagged-dependencies)
+  * [Engine vs Index](#engine-vs-index)
+  * [Setting up](#setting-up)
+  * [Installation](#installation)
+    * [Install the complete SDK run:](#install-the-complete-sdk-run)
+    * [Updates](#updates)
+    * [🛼 Roll-your-own installation](#-roll-your-own-installation)
+  * [Specify environment variables](#specify-environment-variables)
+    * [Understanding your engine prefix and suffix:](#understanding-your-engine-prefix-and-suffix)
+  * [Indexing your CMS content](#indexing-your-cms-content)
+  * [Get search to your users](#get-search-to-your-users)
+<!-- TOC -->
+
+## Tagged dependencies
+
+As part of the Silverstripe Search managed service, we will put the required dependencies (and the explicit versions) through our internal regression testing before adding them to this SDK.
+
+This means that new tags for the SDK **will be slower** than new tags being made on these dependencies. The dependencies (like Forager and Discoverer) are open source/community modules, so they can have contributes coming in faster than we can regression test.
+
+**If you want updates to dependencies faster**, then you should define your own requirements for those dependencies with your own version constraints. By doing so, you accept the risks of your defined versioned constraints.
+
+## Engine vs Index
+
+> [!IMPORTANT]
+> **TL;DR:**\
+> For all intents and purposes, "engine" and "index" are synonomous. If we refer to something as "engine", but the Discoverer module is asking for an "index", then you simply need to give it the data you have for your engine.
+
+The Discoverer module is built to be service-agnostic; meaning, you can use it with any search provider, as long as
+there is an adaptor (like this module) for that service.
+
+When Discoverer refers to an "index", it is talking about the data store used for housing your content. These data
+stores are known by different names across different search providers. Algolia and Elasticsearch call them "indexes",
+Typesense calls them "collections", App Search calls them "engines". Discoverer had to call them **something** in its
+code, and it chose to call then "indexes"; Silverstripe Search, however, calls them "engines".
```

Comment:
worth calling out 'indexing' is the process of adding something to an engine?

Reply:
@blueo do you think something like:

> You will also come across the words "indexing" and "indexed", these is not talking about the data store, these are talking about the _process_ or _outcome_ of adding content into your data store.

Reply:
maybe:

> Actions apply in the same way to all of the above. So in Silverstripe Search the action of `indexing` is adding data to your engine where it is said to be `indexed`. Updating that data is commonly referred to as `re-indexing`. ?

Reply:
Awesome! Thanks! Updated slightly, but have pushed an update to `main` :)